apps/ex_wire/lib/ex_wire/packet/warp_status.ex

```diff
@@ -0,0 +1,180 @@
+defmodule ExWire.Packet.WarpStatus do
+  @moduledoc """
+  Status messages updated to handle warp details.
+
+  ```
+  **Status** [`+0x00`: `P`, `protocolVersion`: `P`, `networkId`: `P`,
+              `td`: `P`, `bestHash`: `B_32`, `genesisHash`: `B_32`,
+              `snapshot_hash`: B_32, `snapshot_number`: P]
+
+  In addition to all the fields in eth protocol version 63’s status (denoted
+  by `...`), include `snapshot_hash` and `snapshot_number` which signify the
+  snapshot manifest RLP hash and block number respectively of the peer's local
+  snapshot.
+  ```
+  """
+
+  require Logger
+
+  @behaviour ExWire.Packet
+
+  @type t :: %__MODULE__{
+          protocol_version: integer(),
+          network_id: integer(),
+          total_difficulty: integer(),
+          best_hash: binary(),
+          genesis_hash: binary(),
+          snapshot_hash: EVM.hash(),
+          snapshot_number: integer()
+        }
+
+  defstruct [
+    :protocol_version,
+    :network_id,
+    :total_difficulty,
+    :best_hash,
+    :genesis_hash,
+    :snapshot_hash,
+    :snapshot_number
+  ]
+
+  @doc """
+  Build a WarpStatus packet
+
+  Note: we are currently reflecting values based on the packet received, but
+  that should not be the case. We should provide the total difficulty of the
+  best chain found in the block header, the best hash, and the genesis hash of
+  our blockchain.
+
+  TODO: Don't parrot the same data back to sender
```

Comment:
This `TODO` also appears [here](https://github.com/mana-ethereum/mana/pull/632/files#diff-53e9c539a86c0c1e93acf699b41c7fb4R59). Does it make sense to have it only in one place?